### DISCUSSION TITLE
GitHub desktop corrupting files.

### DISCUSSION THREAD

**Discussion by JakubOwadowski** [community/community#137540] [Other Feature Feedback, Questions, & Ideas] [answered]
I have an issue. I work with a repository that includes .dds files. Each time I make a commit (it doesn't even include these files), they get corrupted and I have to manually replace them. I know that this is not a hardware issue as I have this issue on two unrelated computers and now testing on a completely new machine (freshly assembled).

Any idea why is it like this?

Edit: I checked and the issue is present with pretty much any graphical asset - .tga files have sections randomly moved and .jpg files change colouration in the middle. Only PNGs seems unaffected.

Maybe it's repo settings,?

**Comment by ThisIs-Developer** [comment, marked as answer]
The corruption of your .dds, .tga, and .jpg files during commits is likely due to Git treating them as text files. Here are key solutions to resolve the issue:

1. **Use a .gitattributes File**: Create or modify a `.gitattributes` file in your repository to specify that these files are binary. Add the following lines:

   ```
   *.dds binary
   *.tga binary
   *.jpg binary
   ```

2. **Consider Git LFS**: If your graphical assets are large, use Git Large File Storage (LFS) to manage them more effectively. Track your file types with:

   ```bash
   git lfs track "*.dds"
   git lfs track "*.tga"
   git lfs track "*.jpg"
   ```

3. **Check .gitignore**: Ensure your `.gitignore` file is not excluding or misconfiguring the handling of these file types.

**Reply by JakubOwadowski** [reply]
I actually managed to fix it myself. 

Curious thing, I had it set that all text files are to be changed to LF... And it randomly decided these were text files as well.

**Reply by ThisIs-Developer** [reply]
Glad to hear you fixed it!😛
